src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -129,6 +246,172 @@ private void ParseMarkupText(in SyntaxListBuilder<RazorSyntaxNode> builder)
             AcceptAndMoveNext();
         }
 
+        private void ParseMarkupInCodeBlock(in SyntaxListBuilder<RazorSyntaxNode> builder)
+        {
+            do
+            {
+                switch (GetParserState(ParseMode.MarkupInCodeBlock))
+                {
+                    case ParserState.EOF:
+                        CompleteMarkupInCodeBlock(builder);
+                        return;
```

Comment:
I think this is worth a comment, since there's cleanup code outside of the switch, but we're explicitly short-circuiting that.